.github/workflows/containerd.yml

```diff
@@ -298,27 +298,38 @@ jobs:
           set -o errexit
           set -o nounset
           set -o pipefail
+          set -x
 
-          BDIR="$(mktemp -d -p $PWD)"
+          BDIR="/var/lib/containerd-critest"
           echo "containerd temp dir: ${BDIR}"
-          mkdir -p ${BDIR}/{root,state}
+          sudo mkdir -p ${BDIR}/{root,state}
 
-          cat > ${BDIR}/config.toml <<EOF
+          sudo bash -c 'cat > ${BDIR}/config.toml <<EOF
           version = 2
           [plugins]
             [plugins.cri.containerd.default_runtime]
               runtime_type = \"${{matrix.runtime}}\"
-          EOF
+          EOF'
 
           # Remove possibly existing containerd configuration
           sudo rm -rf /etc/containerd
 
-          sudo PATH=$PATH /usr/local/bin/containerd -a ${BDIR}/c.sock -root ${BDIR}/root -state ${BDIR}/state -log-level debug &> ${BDIR}/containerd-cri.log &
+          # UserNamespaces only work for 1.7 and main as well as crun
+          # For runc, see:
+          # - https://github.com/opencontainers/runc/issues/4114
+          # - https://github.com/opencontainers/runc/pull/3963
+          SKIP=
+          if [[ "${{ format(matrix.version, '/') }}" == "release/1.6" || "${{matrix.runc}}" == runc ]]; then
+            SKIP=--ginkgo.skip=UserNamespaces
```

Comment:
Shall we parse the output of `<runtime> features` (being runtime crun or runc) with `jq` and if idmap isn't enabled, we skip it? This way, it will automatically run the tests when we use a runc version that works.

Is not a big deal to remove this when we switch to a runc version that supports it

Reply:
Sure, I changed it as suggested :+1:

Reply:
Hm, the features command does not seem to be supported by every runc/crun version we use in CI, which makes things a bit tricky: https://github.com/kubernetes-sigs/cri-tools/actions/runs/7986230493/job/21806204360?pr=1354

I'm going back to the previous approach for now.

Reply:
Ouch, sorry, didn't know that. Thanks!

Reply:
I can't mark this as solved, but it is of course solved :)